pafy/playlist.py

```diff
@@ -309,6 +328,93 @@ def __getitem__(self, index):
         self._items = items
         return self._items[index]
 
+    def __setitem__(self, index, value):
```

Comment:
Why implement `__setitem__`? Is it used somewhere else? I can't think of a reason why you would want it in the public API, so if this functionality is needed internally this should probably be done a different way.

Reply:
To use `random.shuffle()` on `Playlist` object. I think it is reasonable to have that functionality.

Reply:
Hm, I'm not so sure. It seems to me like `Playlist` should be immutable, if you want something that can be sorted you can use `list(playlist)` to get a list from it.

This is also a lot of duplicated code, although so is `__getitem__`...

Reply:
Should I remove `__setitem__`?

I noticed the duplication of code too. But there is not much we can do to reduce that.